packages/openstef-core/src/openstef_core/feature_engineering/temporal_transforms/datetime_features.py

```diff
@@ -0,0 +1,153 @@
+# SPDX-FileCopyrightText: 2025 Contributors to the OpenSTEF project <short.term.energy.forecasts@alliander.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+"""Transform for extracting datetime-based features from time series data.
+
+This module provides functionality to compute datetime features that indicate
+temporal patterns like weekdays, weekends, and other time-based characteristics
+from the datetime index of time series datasets.
+"""
+
+import numpy as np
+import pandas as pd
+
+from openstef_core.datasets import TimeSeriesDataset
+from openstef_core.datasets.transforms import TimeSeriesTransform
+
+MIN_WEEKEND_IDX: int = 5  # Saturday
+SUNDAY_IDX: int = 6
+
+
+class DatetimeFeatures(TimeSeriesTransform):
+    """Transform that adds datetime features to time series data.
+
+    Computes features that are derived from the datetime index of the dataset.
+
+    The features added are:
+        - is_week_day: 1 if the day is a weekday (Monday to Friday),
+            0 otherwise (Saturday or Sunday).
+        - is_weekend_day: 1 if the day is a weekend day (Saturday or Sunday),
+            0 otherwise (Monday to Friday).
+        - is_sunday: 1 if the day is a Sunday, 0 otherwise.
+        - month_of_year: Month of the year (1 to 12).
+        - quarter_of_year: Quarter of the year (1 to 4).
+
+    Example:
+        >>> import pandas as pd
+        >>> from datetime import timedelta
+        >>> from openstef_core.datasets import TimeSeriesDataset
+        >>> from openstef_core.feature_engineering.temporal_transforms.datetime_features import (
+        ...     DatetimeFeatures
+        ... )
+        >>>
+        >>> # Create sample dataset
+        >>> data = pd.DataFrame({
+        ...     'load': [100, 120, 110]
+        ... }, index=pd.date_range('2025-01-01', periods=3,
+        ... freq='D'))
+        >>> dataset = TimeSeriesDataset(data, sample_interval=timedelta(minutes=15))
+        >>> transform = DatetimeFeatures()
+        >>> transformed_dataset = transform.fit_transform(dataset)
+        >>> transformed_dataset.feature_names
+        ['load', 'is_week_day', 'is_weekend_day', 'is_sunday', 'month_of_year', 'quarter_of_year']
+        >>> transformed_dataset.data["is_week_day"].tolist()
+        [1, 1, 1]
+        >>> transformed_dataset.data["month_of_year"].tolist()
+        [1, 1, 1]
+    """
+
+    def __init__(self) -> None:
+        """Initialize the DatetimeFeatures transform."""
+        self.datetime_features: pd.DataFrame = pd.DataFrame()
+
+    @staticmethod
+    def _is_weekend_day(index: pd.DatetimeIndex) -> np.ndarray:
+        """Check if the day is a weekend day (Saturday or Sunday).
+
+        Args:
+            index: DatetimeIndex to check for weekend days.
+
+        Returns:
+            Numpy array where 1 indicates a weekend day and 0 otherwise.
+        """
+        return (index.weekday >= MIN_WEEKEND_IDX).astype(int)
```

Comment:
Similar here. There functions might be overkill since they will probably never be overriden.